### ISSUE TITLE
Close/dismiss button

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-notification#129]
Add a default close/dismiss button to all notifications (a new themable part), to avoid the need for application develpers to do that manually. It is considered a best practice to have an explicit dismiss button for notifications.

Add a new theme variant that the developer can use to opt-out of (hide) the default close/dismiss button.

Open questions: 

- How do we handle the close/dismiss button label, how is it localized (do we need an i18n property)?
- Should ESC close/dismiss the notification, when the notification itself is focused (see #121)?

**Comment by rolfsmeds** [comment]
- Having a dismiss button rendered by default would be a breaking change, and cause issues in existing applications, many of which implement their own dimiss buttons.

- A dedicated boolean property for having a dismiss button might be a cleaner approach.

**Comment by jouni** [comment]
Right, this could be opt-in initially, and marked to be changed to opt-out in the next major version.
